src/EditorFeatures/Core/Implementation/IntelliSense/SignatureHelp/Controller.Session_ComputeModel.cs

```diff
@@ -150,25 +152,29 @@ private static bool SequenceEquals(IEnumerable<string> s1, IEnumerable<string> s
                 return s1 != null && s2 != null && s1.SequenceEqual(s2);
             }
 
-            private static SignatureHelpItem GetSelectedItem(Model currentModel, SignatureHelpItems items, ISignatureHelpProvider provider)
+            private static (SignatureHelpItem selectedItem, bool userSelected) GetSelectedItem(Model currentModel, SignatureHelpItems items, ISignatureHelpProvider provider)
             {
                 // Try to find the most appropriate item in the list to select by default.
 
-                // If the provider specified one a selected item, then always stick with that one. 
-                if (items.SelectedItemIndex.HasValue)
+                // If it's the same provider as the previous model we have, and we had a user-selection,
+                // then try to return the user-selection.
+                if (currentModel != null && currentModel.Provider == provider && currentModel.UserSelected)
                 {
-                    return items.Items[items.SelectedItemIndex.Value];
+                    var userSelected = items.Items.FirstOrDefault(i => DisplayPartsMatch(i, currentModel.SelectedItem));
+                    if (userSelected != null)
+                    {
+                        return (userSelected, true);
+                    }
                 }
 
-                // If the provider did not pick a default, and it's the same provider as the previous
-                // model we have, then try to return the same item that we had before. 
-                if (currentModel != null && currentModel.Provider == provider)
+                // If the provider specified a selected item, then pick that one.
+                if (items.SelectedItemIndex.HasValue)
                 {
-                    return items.Items.FirstOrDefault(i => DisplayPartsMatch(i, currentModel.SelectedItem)) ?? items.Items.First();
+                    return (items.Items[items.SelectedItemIndex.Value], false);
                 }
 
                 // Otherwise, just pick the first item we have.
-                return items.Items.First();
+                return (items.Items.First(), false);
```

Comment:
nit: i'm on the fence if we should use a named param for true/false in this method.  totally up to you.

Reply:
Changed to out parameter